### DISCUSSION TITLE
While building on MacOs Sonoma on M2 Macbook

### DISCUSSION THREAD

**Discussion by trignomtry** [Floorp-Projects/Floorp#856] [Q&A]
### Pre-Submission Checklist

- [x] I have checked the existing/closed issues to ensure this issue has not already been reported.
- [X] I am using the latest supported version of the browser.
- [X] I have tried disabling all browser extensions/add-ons to rule out interference from these.
- [X] I have tried clearing the browser cache and cookies.
- [X] I have tried to create a new profile and the issue still occurs.
- [X] I have provided detailed step-by-step instructions on how to reproduce the issue.
- [X] I have included relevant screenshots or console outputs.
- [X] I have checked this problem is not ocurring in the latest version of Firefox ESR.
- [X] This issue is specific to this browser and does not occur in other browsers.

### Bug Summary

While building on my M2 Macbook Air running Sonoma, I get this error every time when running ./mach build: 

0:07.76 checking the host C++ compiler version... 14.0.0.or.more
 0:07.76 checking the host C++ compiler works... yes
 0:11.62 checking for target linker...
 0:11.62 DEBUG: Executing: `/usr/bin/clang -isysroot /Users/trignomtry/.mozbuild/MacOSX13.3.sdk -mmacosx-version-min=11.0 -std=gnu99 --target=arm64-apple-darwin -Wl,--version`
 0:11.62 ERROR: Failed to find an adequate linker
*** Fix above errors and then restart with "./mach build"

I have tried looking at every forum and have not found a good response. 

### Reproduction Steps

1. Follow all instruction steps, run ./mach build in root directory.
2. run ./mach build in root directory.

### Expected Behavior

I expect it to find a linker automatically

### Actual Behavior

The system fails to find an adequate linker and build fails. 

### Screenshots

<img width="1093" alt="Screenshot 2024-02-05 at 8 08 00 PM" src="https://github.com/Floorp-Projects/Floorp/assets/106871964/7b244f96-221f-4151-b35f-aecaa14126c7">

### Browser Version

Letest

### Operating System

Sonoma 14.2

**Comment by Ryosuke-Asano** [comment]
Please do not create build-related Issues. We will not respond.

**Comment by Ryosuke-Asano** [comment]
It's means... Clang or another build tools's version is too newer or old
